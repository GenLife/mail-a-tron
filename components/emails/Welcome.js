import React from 'react'
import Layout from '../Layout'
import { Image, Item, Span, A } from 'react-html-email'
import Button from '../Button'

const rowStyle = {
  width: '300px'
}

// Sets maximum width for each row of content
export default ({ name }) => (
  <Layout>

    {/* Image subtitle Row */}

    <Item
      align='center'
      valign='top'
      style={{ ...rowStyle, paddingBottom: '30px' }}
        >
      <Span
        color='#444444'
        fontSize={25}
      >
        {`Welcome${name ? ' ' + name : ''}!`}
      </Span>
    </Item>

    {/* Video Row */}

    <Item
      align='center'
      valign='top'
      style={rowStyle}
        >
      <A href='http://gen.life'>
        <Image
          data-mc-bar='bar'
          data-mc-baz='baz'
          alt='video-image'
          src='static/video.jpg'
          width={300}
          height={240}
          vspace={20}
            />
      </A>
    </Item>

    {/* Image subtitle Row */}

    <Item
      align='center'
      valign='top'
      style={{ ...rowStyle, padding: '25px 0px' }}
        >
      <Span
        color='#444444'
        fontSize={25}
          >
            Our Greatest Feature Ever
          </Span>
    </Item>

    {/* Text  Row */}

    <Item
      align='center'
      valign='top'
      style={rowStyle}
        >
      <Span
        fontSize={14}
        color='#444444'
        lineHeight={20}
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed varius, leo a ullamcorper feugiat, ante purus sodales justo, a faucibus libero lacus a est. Aenean at mollis ipsum.
          </Span>
    </Item>

    {/* Button Row */}

    <Item style={rowStyle}>
      <Button href={'http://gen.life'} />
    </Item>
  </Layout>
)
