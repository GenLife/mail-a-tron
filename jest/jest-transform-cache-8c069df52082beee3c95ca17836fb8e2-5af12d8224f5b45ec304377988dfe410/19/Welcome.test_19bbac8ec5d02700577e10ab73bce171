'use strict';
var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _Welcome = require('./Welcome');var _Welcome2 = _interopRequireDefault(_Welcome);
var _enzyme = require('enzyme');
var _enzymeToJson = require('enzyme-to-json');var _enzymeToJson2 = _interopRequireDefault(_enzymeToJson);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };} /* global it, expect, describe */

describe('Welcome', function () {
  it('it renders normally', function () {
    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Welcome2.default, null));
    expect((0, _enzymeToJson2.default)(wrapper)).toMatchSnapshot();
  });

  it('it renders with a name', function () {
    var wrapper = (0, _enzyme.shallow)(_react2.default.createElement(_Welcome2.default, { name: 'Jo\xE3o' }));
    expect((0, _enzymeToJson2.default)(wrapper)).toMatchSnapshot();
  });
});