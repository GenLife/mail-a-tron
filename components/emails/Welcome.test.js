/* global it, expect, describe */
import React from 'react'
import Welcome from './Welcome'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Welcome', () => {
  it('it renders normally', () => {
    const wrapper = shallow(<Welcome />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('it renders with a name', () => {
    const wrapper = shallow(<Welcome name='JoÃ£o' />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
