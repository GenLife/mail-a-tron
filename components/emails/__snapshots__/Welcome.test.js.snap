// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome it renders normally 1`] = `
<Component>
  <Item
    align="center"
    style={
      Object {
        "paddingBottom": "30px",
        "width": "300px",
      }
    }
    valign="top"
  >
    <Span
      color="#444444"
      fontFamily="sans-serif"
      fontSize={25}
    >
      Welcome!
    </Span>
  </Item>
  <Item
    align="center"
    style={
      Object {
        "width": "300px",
      }
    }
    valign="top"
  >
    <A
      href="http://gen.life"
      textDecoration="underline"
    >
      <Image
        alt="video-image"
        data-mc-bar="bar"
        data-mc-baz="baz"
        height={240}
        src="static/video.jpg"
        vspace={20}
        width={300}
      />
    </A>
  </Item>
  <Item
    align="center"
    style={
      Object {
        "padding": "25px 0px",
        "width": "300px",
      }
    }
    valign="top"
  >
    <Span
      color="#444444"
      fontFamily="sans-serif"
      fontSize={25}
    >
      Our Greatest Feature Ever
    </Span>
  </Item>
  <Item
    align="center"
    style={
      Object {
        "width": "300px",
      }
    }
    valign="top"
  >
    <Span
      color="#444444"
      fontFamily="sans-serif"
      fontSize={14}
      lineHeight={20}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed varius, leo a ullamcorper feugiat, ante purus sodales justo, a faucibus libero lacus a est. Aenean at mollis ipsum.
    </Span>
  </Item>
  <Item
    style={
      Object {
        "width": "300px",
      }
    }
  >
    <Component
      href="http://gen.life"
    />
  </Item>
</Component>
`;

exports[`Welcome it renders with a name 1`] = `
<Component>
  <Item
    align="center"
    style={
      Object {
        "paddingBottom": "30px",
        "width": "300px",
      }
    }
    valign="top"
  >
    <Span
      color="#444444"
      fontFamily="sans-serif"
      fontSize={25}
    >
      Welcome Jo√£o!
    </Span>
  </Item>
  <Item
    align="center"
    style={
      Object {
        "width": "300px",
      }
    }
    valign="top"
  >
    <A
      href="http://gen.life"
      textDecoration="underline"
    >
      <Image
        alt="video-image"
        data-mc-bar="bar"
        data-mc-baz="baz"
        height={240}
        src="static/video.jpg"
        vspace={20}
        width={300}
      />
    </A>
  </Item>
  <Item
    align="center"
    style={
      Object {
        "padding": "25px 0px",
        "width": "300px",
      }
    }
    valign="top"
  >
    <Span
      color="#444444"
      fontFamily="sans-serif"
      fontSize={25}
    >
      Our Greatest Feature Ever
    </Span>
  </Item>
  <Item
    align="center"
    style={
      Object {
        "width": "300px",
      }
    }
    valign="top"
  >
    <Span
      color="#444444"
      fontFamily="sans-serif"
      fontSize={14}
      lineHeight={20}
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed varius, leo a ullamcorper feugiat, ante purus sodales justo, a faucibus libero lacus a est. Aenean at mollis ipsum.
    </Span>
  </Item>
  <Item
    style={
      Object {
        "width": "300px",
      }
    }
  >
    <Component
      href="http://gen.life"
    />
  </Item>
</Component>
`;
